File: client/app/src/main/java/com/rh/heji/network/request/CategoryEntity.java
package com.rh.heji.network.request;

import com.rh.heji.data.db.Category;

/**
 * Date: 2020/9/24
 * Author: 锅得铁
 * #标签
 */
public class CategoryEntity {

    private String name;

    private Integer type;

    private Integer level;

    public CategoryEntity(Category category) {
        this.name = category.getCategory();
        this.type = category.getType();
        this.level = category.getLevel();
    }

    public Category toDbCategory() {
        Category category = new Category(name, level, type);
        return category;
    }

    public Integer getType() {
        return type;
    }

    public void setType(Integer type) {
        this.type = type;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Integer getLevel() {
        return level;
    }

    public void setLevel(Integer level) {
        this.level = level;
    }
}

File: client/app/src/main/java/com/rh/heji/network/request/CategoryEntity.kt
package com.rh.heji.network.request

import com.rh.heji.data.db.Category

/**
 * Date: 2020/9/24
 * @author: 锅得铁
 * #标签
 */

data class CategoryEntity(val category: Category) {
    var id: String = category.id
    var bookId: String = category.bookId
    var name: String = category.name
    var type: Int = category.type
    var level: Int = category.level
    fun toDbCategory(): Category {
        return Category(bookId, name).apply {
            id = category.id
            level = category.level
            type = category.type
        }
    }

}